### ISSUE TITLE
[a11y]:  DataTable [selection] - "aria-label" is not valid for the element <label>

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14053]
### Package

carbon-components-react

### Browser

Chrome

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset

Accessibility Checker

### Description

![image (1)](https://github.com/carbon-design-system/carbon/assets/32720851/2168d26d-cdb8-4c35-b223-ea27fc5823d1)
![image (2)](https://github.com/carbon-design-system/carbon/assets/32720851/1e109dce-e53c-4e45-b94f-5bac5b4aa70a)

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-selection--default

### Steps to reproduce

Run a11y checker on the above story

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
